src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1292,6 +1291,64 @@ public void ShowUpdatePackages(ShowUpdatePackageOptions updatePackageOptions)
                     SelectMatchingUpdatePackages(updatePackageOptions);
                 };
                 _packageList.LoadItemsCompleted += handler;
+            }  
+        }
+
+        public void SelectFirstPackage(string packageVersion)
+        {
+            _topPanel.SelectFilter(ItemFilter.All);
+            SelectedSource = PackageSources.FirstOrDefault(psm => psm.IsAggregateSource);
+            EventHandler handler = null;
+            handler = (s, e) =>
+            {
+                _packageList.LoadItemsCompleted -= handler;
+
+                _packageList.UpdateSelectedItem((PackageItemViewModel)_packageList.Items.FirstOrDefault());
+                if (_packageList.SelectedItem == null)
```

Comment:
What's the scenario where the first item in the list would be null after the list is finished populating but we want to hook up an event handler for when the selection changes to then find the exact package match?

Reply:
There seems to be some inconsistent behavior when I call UpdatedSelectedItem, sometimes it will load in the first package fine before I perform work on it while other times the code will try to perform work on the first package before it's fully loaded. My guess is that there is some kind of race condition here and using this event handler case was the way I went about to take care of both cases mentioned above.

Reply:
How could it try to do work before the package is loaded when this event is supposed to only fire after the package is loaded?

Reply:
I have no idea how, but it would do that in some of my previous testing of the feature, so I had to mitigate it by doing the event handler